src/mercury_engine_data_structures/formats/collision.py

```diff
@@ -1,45 +1,49 @@
 from __future__ import annotations
 
+import copy
+
 import construct
 from construct import Array, Container, Flag, Hex, Int8ul, Int16ul, Rebuild, Struct, Switch
 
 from mercury_engine_data_structures import game_check
 from mercury_engine_data_structures.common_types import (
     CVector2D,
     CVector3D,
+    CVector4D,
     Float,
     StrId,
     UInt,
+    Vec2,
+    Vec3,
+    Vec4,
     make_vector,
 )
 from mercury_engine_data_structures.construct_extensions.misc import ErrorWithMessage, OptionalValue
-from mercury_engine_data_structures.game_check import Game
 
 CollisionPoint = Struct(
-    "x" / Float,
-    "y" / Float,
+    "position" / CVector2D,
     "material_attribute" / UInt,
 )
 CollisionPolySR = Struct(
     "num_points" / Rebuild(UInt, construct.len_(construct.this.points)),
     "unk4" / Hex(construct.Byte),
     "unk5" / Hex(UInt),
     "points" / Array(construct.this.num_points, CollisionPoint),
-    "boundings" / Array(4, Float),
+    "boundings" / CVector4D,
 )
 CollisionPolyDread = Struct(
     "num_points" / Rebuild(UInt, construct.len_(construct.this.points)),
     "unk" / Hex(UInt),
     "points" / Array(construct.this.num_points, CollisionPoint),
     "loop" / Flag,
-    "boundings" / Array(4, Float),
+    "boundings" / CVector4D,
```

Comment:
```py
def calculate_poly_boundings(ctx: Container) -> Container:
    x1 = min(point.x for point in ctx.points)
    y1 = min(point.y for point in ctx.points)
    x2 = max(point.x for point in ctx.points)
    y2 = max(point.y for point in ctx.points)
    return Container({"min": Vec2(x1, y1), "max": Vec2(x2, y2)})
```
and then
`"boundings" / Rebuild(BoundingBox2D, calculate_poly_boundings)` for both this and the SR construct